backend/src/router.rs

```diff
@@ -0,0 +1,404 @@
+            .map(|path_item_type| check_api::<_, _, _, H, Version, Doc>(path, &path_item_type))
+        {
+            Ok(Ok(())) => (),
+            Ok(Err(err)) | Err(err) => {
+                if let Some(errors) = &mut self.api_errors {
+                    errors.extend_one(err);
+                } else {
+                    self.api_errors = Some(err);
+                }
+            }
+        }
+
+        match try_convert_method_filter_from_method(method) {
+            Ok(method) => self.inner = self.inner.route(path, on(method, handler)),
+            Err(err) => {
+                if let Some(errors) = &mut self.api_errors {
+                    errors.extend_one(err);
+                } else {
+                    self.api_errors = Some(err);
+                }
+            }
+        };
+
+        self
+    }
+}
+
+impl<'a, Version, Doc, S, B> Default for ContextRouter<Version, Doc, S, B>
+where
+    Version: ApiVersion<'a>,
+    Doc: OpenApi,
+    B: HttpBody + Send + 'static,
+    S: Clone + Send + Sync + 'static,
+{
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<V, D, S, B> Deref for ContextRouter<V, D, S, B> {
+    type Target = Router<S, B>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+pub trait RouterApiExt<S = (), B = Body, E = Infallible> {
+    /// Wraps `Router` with `ApiVersion` and `OpenApi` instances into the new context to call
+    /// `api_route` with said context
+    fn with_context<Version, Doc>(self) -> ContextRouter<Version, Doc, S, B>;
+}
+
+impl<S, B> RouterApiExt<S, B, Infallible> for Router<S, B>
+where
+    B: HttpBody + Send + 'static,
+    S: Clone + Send + Sync + 'static,
+{
+    fn with_context<Version, Doc>(self) -> ContextRouter<Version, Doc, S, B> {
+        ContextRouter {
+            inner: self,
+            context: PhantomData,
+            api_errors: None,
+        }
+    }
+}
+
+/// Check if the following route corresponds with `OpenAPI` declaration
+/// Relies on `operation_id` field, must NOT be changed on handler's declaration
+#[cfg(all(feature = "swagger", debug_assertions))]
+fn check_api<'a, T, S, B, H, Version, ApiDocumentation>(
+    path: &str,
+    method: &PathItemType,
+) -> Result<(), RouteError>
+where
+    H: Handler<T, S, B>,
+    T: 'static,
+    S: Send + Sync + 'static,
+    ApiDocumentation: OpenApi,
+    Version: ApiVersion<'a>,
+{
+    let route = [
+        Version::to_path(),
+        &path
+            .split('/')
+            .map(|arg| {
+                arg.starts_with(':')
+                    .then(|| ["{", &arg[1..], "}"].concat())
+                    .unwrap_or_else(|| arg.to_string())
+            })
+            .collect::<Vec<_>>()
+            .join("/"),
+    ]
+    .concat();
+    let operation_id = ApiDocumentation::openapi()
+        .paths
+        .get_path_item(&route)
+        .ok_or(RouteError::NoRoute)
+        .attach_printable(format!("route: {route}"))?
```

Comment:
Right now on every call the new `String` is created here (even when there is no error here). To prevent this it is better to use `attach_printable_lazy()`